lang/LangSource/GC.h

```diff
@@ -82,51 +84,47 @@ class PyrGC
 
 	MALLOC PyrObject* NewFinalizer(ObjFuncPtr finalizeFunc, PyrObject *inObject, bool inCollect);
 
-	bool IsBlack(PyrObjectHdr* inObj)  { return inObj->gc_color == mBlackColor; }
-	bool IsWhite(PyrObjectHdr* inObj)  { return inObj->gc_color == mWhiteColor; }
-	bool IsGrey(PyrObjectHdr* inObj)   { return inObj->gc_color == mGreyColor; }
-	static bool IsMarker(PyrObjectHdr* inObj) { return inObj->gc_color == obj_gcmarker; }
-	bool IsFree(PyrObjectHdr* inObj)   { return (!(IsMarker(inObj) ||
+	BOOST_FORCEINLINE bool IsBlack(PyrObjectHdr* inObj)  { return inObj->gc_color == mBlackColor; }
```

Comment:
functions defined within a class are inline functions, so the `inline` keyword would not change anything. nevertheless, `inline` is only a hint for the compiler and the compiler is free not to inline them. in the case of the sclang bytecode interpreter this happened, probably because of function growth constraints.

fwiw, compilers may have good reasons not to inline certain functions and usually they do a good job (gcc is better, msvc is worse). but if they don't we can force them to do so. unfortunately controlling the inlining from the caller side is only implemented by gcc (flatten attribute), neither clang nor msvc support it.